packages/playwright-test/src/docker/docker.ts

```diff
@@ -96,17 +96,29 @@ interface ContainerInfo {
 }
 
 export async function containerInfo(): Promise<ContainerInfo|undefined> {
-  const containerId = await findRunningDockerContainerId();
-  if (!containerId)
+  const container = await findRunningDockerContainer();
+  if (!container)
     return undefined;
-  const logLines = await dockerApi.getContainerLogs(containerId);
+  const vncBinding = container.portBindings.find(binding => binding.containerPort === 7900);
+  const wsBinding = container.portBindings.find(binding => binding.containerPort === 5400);
+  if (!vncBinding || !wsBinding)
+    return undefined;
+  const logLines = await dockerApi.getContainerLogs(container.containerId);
+
+  const containerUrlToHostUrl = (address: string, portBinding: dockerApi.PortBinding) => {
+    const url = new URL(address);
+    url.host = portBinding.ip;
+    url.port = portBinding.hostPort + '';
```

Comment:
```suggestion
    url.port = container.portBindings.find(binding => biding.containerPort === +url.port).hostPort + '';
```

This way you don't need to hardcode 5400 and 7900.